include/internal/catch_generators.hpp

```diff
@@ -16,8 +16,22 @@
 #include <cassert>
 
 #include <utility>
+#include <exception>
 
 namespace Catch {
+
+class GeneratorException : public std::exception {
+    const char* const m_msg = "";
+
+public:
+    GeneratorException(const char* msg):
+        m_msg(msg)
+    {}
+
+
+    const char* what() const noexcept override;
```

Comment:
I think this can final too unless there will be more exception types deriving from `GeneratorException`.

Reply:
Even if there are, we can always remove it for our use. The question then is, whether it should be a user-facing customization point and I think the answer is "no".